service/src/test/java/com/ge/predix/acs/attribute/connectors/DefaultSubjectAttributeStoreTest.java

```diff
@@ -0,0 +1,70 @@
+package com.ge.predix.acs.attribute.connectors;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.ge.predix.acs.privilege.management.PrivilegeManagementService;
+import com.ge.predix.acs.rest.BaseSubject;
+
+@Test
+public class DefaultSubjectAttributeStoreTest {
+    @Mock
+    private PrivilegeManagementService privilegeManagementService;
+
+    @Autowired
+    @InjectMocks
+    private DefaultSubjectAttributeStore defaultSubjectAttributeStore;
+
+    @BeforeMethod
+    public void beforeMethod() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testGetAttributesForNonExistentSubject() throws Exception {
+        Mockito.when(this.privilegeManagementService.getBySubjectIdentifierWithInheritedAttributes(Mockito.any()))
+               .thenReturn(null);
+        Assert.assertTrue(this.defaultSubjectAttributeStore.getAttributes("nonexistentSubject").isEmpty());
+    }
+
+    @Test
+    public void testGetAttributesByCriteriaForNonExistentSubject() throws Exception {
+        Mockito.when(this.privilegeManagementService.getBySubjectIdentifierAndScopes(Mockito.any(), Mockito.any()))
+               .thenReturn(null);
+        Assert.assertTrue(this.defaultSubjectAttributeStore.getAttributes("nonexistentSubject").isEmpty());
+    }
+
+    @Test(expectedExceptions = ClassCastException.class)
+    public void testGetAttributesByInvalidCriteria() throws Exception {
+        BaseSubject subject = new BaseSubject("someSubject");
+        Mockito.when(this.privilegeManagementService.getBySubjectIdentifierAndScopes(Mockito.any(), Mockito.any()))
+               .thenReturn(subject);
+        Set<String> criteria = Collections.singleton("someCriteria");
+        this.defaultSubjectAttributeStore.getAttributesByCriteria(subject.getSubjectIdentifier(), criteria);
+    }
```

Comment:
This expectedException would change with the new DefaultAttributeStore interface. See above comment